hack/benchmark/assets/x1b/loader.go

```diff
@@ -0,0 +1,195 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package x1b
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+	"unsafe"
+
+	"github.com/vdaas/vald/internal/errors"
+)
+
+const (
+	headerSize = 4
+)
+
+var (
+	ErrOutOfBounds         = errors.New("out of bounds")
+	ErrUnsupportedFileType = errors.New("unsupported file type")
+)
+
+type X1b interface {
+	Load(i int) (interface{}, error)
+	Dimension() int
+	Size() int
+	Close() error
+}
+
+type Bvecs interface {
+	X1b
+	LoadUint8(i int) ([]uint8, error)
+}
+
+type Fvecs interface {
```

Comment:
I think the naming of this interface is not good.
I suggest rename this to `FloatVector`